pyfixest/estimation/decomposition.py

```diff
@@ -3,10 +3,24 @@
 
 import numpy as np
 import pandas as pd
-from joblib import Parallel, delayed
 from scipy.sparse import csc_matrix
 from scipy.sparse.linalg import lsqr
-from tqdm import tqdm
+
+try:
+    from tqdm import tqdm
+except ImportError:
+
+    def tqdm(iterable, *args, **kwargs):
+        "Define a dummy tqdm function."
+        return iterable
+
+
+try:
+    from joblib import Parallel, delayed
+
+    joblib_available = True
+except ImportError:
+    joblib_available = False
```

Comment:
shall we at least raise a warning so that these import errors do not pass silently?

Reply:
done 👍